pkg/kubelet/kubelet.go

```diff
@@ -1040,6 +1040,17 @@ func (kl *Kubelet) pullImageAndRunContainer(pod *api.Pod, container *api.Contain
 		glog.Errorf("Error running pod %q container %q: %v", podFullName, container.Name, err)
 		return "", err
 	}
+	// If the /var/log/containers directory does not exist, create it.
+	if _, err := os.Stat("/var/log/containers"); err != nil {
+		if err := os.Mkdir("/var/log/containers", 0755); err != nil {
+			glog.Errorf("Failed to create directory /var/log/containers: %v", err)
+		}
+	}
+	containerLogFile := fmt.Sprintf("/var/lib/docker/containers/%s/%s-json.log", containerID, containerID)
+	if err = os.Symlink(containerLogFile,
+		fmt.Sprintf("/var/log/containers/%s-%s-%s.log", podFullName, container.Name, containerID)); err != nil {
+		glog.Errorf("Failed to create symbolic link to the log file of container %s/%s: %v", podFullName, container.Name, err)
```

Comment:
I did not remove them when a container dies because they need to live for long enough to have their final contents collected by Fluentd. I wondered about spawing off a goroutine that waited 10 minutes and then removed the link.